vespalib/src/vespa/vespalib/stllike/string.h

```diff
@@ -315,12 +349,12 @@ class small_string
         const char *found = (const char *)memchr(buf, c, _sz-start);
         return (found != NULL) ? (found - buffer()) : (size_type)npos;
     }
-    small_string & assign(const char * s) { return assign(s, strlen(s)); }
-    small_string & assign(const void * s, size_type sz);
-    small_string & assign(stringref s, size_type pos, size_type sz) {
+    small_string & assign(const char * s) noexcept { return assign(s, strlen(s)); }
+    small_string & assign(const void * s, size_type sz) noexcept;
+    small_string & assign(stringref s, size_type pos, size_type sz) noexcept {
         return assign(s.data() + pos, sz);
     }
-    small_string & assign(stringref rhs) {
+    small_string & assign(stringref rhs) noexcept {
```

Comment:
assign-style functions are less commonly `noexcept` since it's expected that the underlying allocation may throw. Since we're running under vespamalloc this shouldn't be a problem (process gets killed instead, which is much easier to reason about), but just a general observation.

Reply:
Agree, but added noexcept for the same reason as you gave here.